From 8fa6d28a7a15ce42950cf75b0869a93389d8d2f9 Mon Sep 17 00:00:00 2001
From: Andreas Robinson <andr345@gmail.com>
Date: Mon, 12 Oct 2009 10:26:58 +0200
Subject: [PATCH 16/19] modprobe: change softdep --pre/--post to pre:/post:

"--" is modified by underscores(). This would have made the
future implementation of softdep in .modinfo more complicated.

Signed-off-by: Andreas Robinson <andr345@gmail.com>
(cherry picked from commit 9127f56dd2d62cdbac99189984286f876b37a704)
---
 modprobe.c                       |  4 ++--
 tests/test-modprobe/28softdep.sh | 10 +++++-----
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/modprobe.c b/modprobe.c
index c270b16..be22875 100644
--- a/modprobe.c
+++ b/modprobe.c
@@ -1144,9 +1144,9 @@ static int parse_config_file(const char *filename,
 			if (!modname || !ptr)
 				goto syntax_error;
 			while ((tk = strsep_skipspace(&ptr, "\t ")) != NULL) {
-				if (streq(tk, "--pre")) {
+				if (streq(tk, "pre:")) {
 					pre = 1; post = 0;
-				} else if (streq(tk, "--post")) {
+				} else if (streq(tk, "post:")) {
 					pre = 0; post = 1;
 				} else if (pre) {
 					pre_modnames = NOFAIL(
diff --git a/tests/test-modprobe/28softdep.sh b/tests/test-modprobe/28softdep.sh
index d999872..234a214 100644
--- a/tests/test-modprobe/28softdep.sh
+++ b/tests/test-modprobe/28softdep.sh
@@ -31,9 +31,9 @@ mkdir -p tests/tmp/etc/modprobe.d
 cat > tests/tmp/etc/modprobe.d/modprobe.conf << EOF
 alias a_alias a
 alias c_alias c
-softdep c --pre a_alias b --post d e
-softdep e --post f
-softdep f --pre a
+softdep c pre: a_alias b post: d e
+softdep e post: f
+softdep f pre: a
 EOF
 
 # Insert-test
@@ -57,8 +57,8 @@ R2=`echo $R1`	# remove newlines
 mkdir -p tests/tmp/etc/modprobe.d
 cat > tests/tmp/etc/modprobe.d/modprobe.conf << EOF
 alias a_alias a
-softdep a --pre b
-softdep b --pre a
+softdep a pre: b
+softdep b pre: a
 EOF
 
 R1=`modprobe -v a_alias 2>&1 | cat` # shell won't assign to R1 without no-op cat.
-- 
1.9.0

